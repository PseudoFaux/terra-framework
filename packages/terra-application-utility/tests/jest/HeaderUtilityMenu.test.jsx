import React from 'react';
import HeaderUtilityMenu from '../../src/header/_HeaderUtilityMenu';

describe('HeaderUtilityMenu', () => {
  const mockOnChange = jest.fn();
  const mockOnDisclose = jest.fn();
  const mockOnRequestClose = jest.fn();
  const config = [{ key: 'key', title: 'title', contentLocation: 'footer', isSelected: false, childKeys: [] }];

  it('should render a header utility menu', () => {
    const wrapper = shallow(
      <HeaderUtilityMenu
        initialSelectedKey="key"
        isHeightBounded={false}
        menuItems={config}
        onChange={mockOnChange}
        onDisclose={mockOnDisclose}
        onRequestClose={mockOnRequestClose}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
