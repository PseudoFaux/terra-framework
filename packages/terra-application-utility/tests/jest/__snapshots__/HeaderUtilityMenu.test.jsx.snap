// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderUtilityMenu it should pass in a custom prop 1`] = `
<ContentContainer
  aria-label="Utility Menu"
  className="utility-menu"
  fill={false}
  footer={null}
  header={
    <div
      className="header"
    >
      <span
        className="left-container"
      >
        <button
          aria-label="Back button"
          className="back-button"
          onClick={[Function]}
        >
          <IconLeft
            className="icon-left"
            data-name="Layer 1"
            isBidi={true}
            viewBox="0 0 48 48"
            xmlns="http://www.w3.org/2000/svg"
          />
        </button>
        <span
          className="header-text"
        >
          title
        </span>
      </span>
      <MenuDivider />
    </div>
  }
  id="test"
  onDisclose={[Function]}
  role="navigation"
/>
`;

exports[`HeaderUtilityMenu should render with default props 1`] = `
<ContentContainer
  aria-label="Utility Menu"
  className="utility-menu"
  fill={false}
  footer={null}
  header={
    <div
      className="header"
    >
      <span
        className="left-container"
      >
        <button
          aria-label="Back button"
          className="back-button"
          onClick={[Function]}
        >
          <IconLeft
            className="icon-left"
            data-name="Layer 1"
            isBidi={true}
            viewBox="0 0 48 48"
            xmlns="http://www.w3.org/2000/svg"
          />
        </button>
        <span
          className="header-text"
        >
          title
        </span>
      </span>
      <MenuDivider />
    </div>
  }
  onDisclose={[Function]}
  role="navigation"
/>
`;

exports[`HeaderUtilityMenu should render with isHeightBounded 1`] = `
<ContentContainer
  aria-label="Utility Menu"
  className="utility-menu"
  fill={true}
  footer={null}
  header={
    <div
      className="header"
    >
      <span
        className="left-container"
      >
        <button
          aria-label="Back button"
          className="back-button"
          onClick={[Function]}
        >
          <IconLeft
            className="icon-left"
            data-name="Layer 1"
            isBidi={true}
            viewBox="0 0 48 48"
            xmlns="http://www.w3.org/2000/svg"
          />
        </button>
        <span
          className="header-text"
        >
          title
        </span>
      </span>
      <MenuDivider />
    </div>
  }
  onDisclose={[Function]}
  role="navigation"
/>
`;

exports[`HeaderUtilityMenu should render with onRequestClose set 1`] = `
<ContentContainer
  aria-label="Utility Menu"
  className="utility-menu"
  fill={false}
  footer={null}
  header={
    <div
      className="header"
    >
      <span
        className="left-container"
      >
        <button
          aria-label="Back button"
          className="back-button"
          onClick={[Function]}
        >
          <IconLeft
            className="icon-left"
            data-name="Layer 1"
            isBidi={true}
            viewBox="0 0 48 48"
            xmlns="http://www.w3.org/2000/svg"
          />
        </button>
        <span
          className="header-text"
        >
          title
        </span>
      </span>
      <MenuDivider />
    </div>
  }
  onDisclose={[Function]}
  role="navigation"
/>
`;
